src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -1736,12 +1737,18 @@ private void MapValuesForPersistence(ContentItemSave contentSave)
             bool Varies(Property property) => property.PropertyType.VariesByCulture();
 
             var variantIndex = 0;
-
+            var newContent = (contentSave.Action == ContentSaveAction.SaveNew
+                              || contentSave.Action == ContentSaveAction.PublishNew
+                              || contentSave.Action == ContentSaveAction.ScheduleNew
+                              || contentSave.Action == ContentSaveAction.SendPublishNew
+                              || contentSave.Action == ContentSaveAction.PublishWithDescendantsNew
+                              || contentSave.Action == ContentSaveAction.PublishWithDescendantsForceNew
+                );
             //loop through each variant, set the correct name and property values
             foreach (var variant in contentSave.Variants)
             {
                 //Don't update anything for this variant if Save is not true
-                if (!variant.Save) continue;
+                if (!variant.Save && !newContent) continue;
```

Comment:
AH, i see the issue

> When we save content the first time, all variants are saved no matter the save flag (Otherwise we loose data, when the path is updated

So basically if you fill out english and french but only mark french to save then you'll lose the english because of the redirect. I feel like we talked about whether we should auto-create all variants or not at one stage and I think the answer was we should (but can't remember why). Maybe @zpqrtbnk can remember? Otherwise a different solution could be in the angular side of things and we auto check all save boxes if the variant is dirty and disable toggling it?

Reply:
The solution you propose will work and is certainly prettier. I can look into it, although it is more frontend

Reply:
Now the functionality is moved to the frontend

![image](https://user-images.githubusercontent.com/1561480/50957850-8ee21100-14bf-11e9-8d43-e5641dbee981.png)